sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// Enables authentication to Azure Active Directory using Azure CLI to generated an access token.
+type AzureCLICredential struct {
+	Client      *azureCLICredentialClient
+	shellClient shellClient
+}
+
+// AzureCLICredentialOptions contains the parameters for the ShellClient to execute the Azure CLI command that will return an Access Token.
+type AzureCLICredentialOption struct {
+	shellClientOption shellClient
```

Comment:
First, *Option should have an 's' suffix.
Second, if the type is an *Options, then why does the field name refer to a client and not to an options? I think the field name should be shellClientOptions.

Reply:
We have a new design,  azAccessTokenProvider implements interface azureCLIAccessTokenProvider would be as the parameter of NewAzureCLICredentialClient. No extra para to NewAzureCLICredential.  
And there are some comments been added to explain the logic.